onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -8372,6 +8372,52 @@ def aten_unique_consecutive(
     raise NotImplementedError()
 
 
+@torch_op("aten::unique", trace_only=True)
+def aten_unique(
+    self: TensorType,
+    sorted: bool = True,
+    return_inverse: bool = False,
+    return_counts: bool = False,
+    dim: Optional[int] = None,
+) -> tuple[TensorType, TensorType, TensorType]:
+    """unique(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False, int? dim=None) -> (Tensor, Tensor?, Tensor?)"""
+
+    if dim is None:
+        unique_values, inverse_indices, counts = aten_unique2(self, sorted, return_inverse, return_counts)
+    else:
+        unique_values, inverse_indices, counts = aten_unique_dim(self, dim, sorted, return_inverse, return_counts)
+    if return_inverse:
+        if return_counts:
+            result = unique_values, inverse_indices, counts
+        else:
+            result = unique_values, inverse_indices
+    elif return_counts:
+        result = unique_values, counts
+    else:
+        result = unique_values
+    return result
+
+
+@torch_op("aten::_unique2")
+def aten_unique2(
+    self: TensorType,
+    sorted: bool = True,
+    return_inverse: bool = False,
+    return_counts: bool = False
+) -> tuple[TensorType, TensorType, TensorType]:
+    """unique(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False) -> (Tensor, Tensor, Tensor)"""
+
+    unique_values, indices, inverse_indices, counts = op.Unique(self, axis=None, sorted=sorted)
+    # HACK: force indices to be in the graph so that it gets a name during optimization
```

Comment:
Is this a bug we should fix elsewhere?

Reply:
I think this could possibly be considered a different bug. The other one is a side-effect of  `onnxscript.optimizer.constant_folding.fold_constants`, whereas this one is a side-effect of the function linked below, which converts the names of unused outputs to empty strings but only removes them if they are trailing. Since `inverse_indices` and `counts` are used, it leads to an error being raised in [`onnxscript.Scope.lookup_or_create`](https://github.com/microsoft/onnxscript/blob/69ae7f421e2f1931e353949fa5e3a0fb23dbe622/onnxscript/_legacy_ir/visitor.py#L194) due to the empty string name given to `indices`.
https://github.com/microsoft/onnxscript/blob/69ae7f421e2f1931e353949fa5e3a0fb23dbe622/onnxscript/optimizer/remove_unused.py#L14